common/test/java/com/couchbase/lite/internal/core/PeerBindingTest.kt

```diff
@@ -0,0 +1,98 @@
+package com.couchbase.lite.internal.core
+
+
+import com.couchbase.lite.internal.core.peers.TaggedWeakPeerBinding
+import org.junit.Assert
+import org.junit.Test
+import java.lang.IllegalStateException
+
+class PeerBindingTest {
+
+    // Binding valid key value pair should create a binding in map
+    @Test
+    fun testCreateBasicPeerBinding() {
+        val binding = TaggedWeakPeerBinding<Any>()
+        try {
+            val key = binding.reserveKey()
+            val mockObject = object {}
+
+            // reserve a key should create a new entry in the map
+            Assert.assertEquals(1, binding.size())
+            Assert.assertEquals(null, binding.getBinding(key))
+
+            binding.bind(key, mockObject)
+            Assert.assertEquals(1, binding.size())
+            Assert.assertEquals(mockObject, binding.getBinding(key))
+
+            // it should be fine to bind the same object and key again
+            binding.bind(key, mockObject)
+            Assert.assertEquals(1, binding.size())
+            Assert.assertEquals(mockObject, binding.getBinding(key))
+        }
+        finally { binding.clear() }
+    }
+
+    // Trying to bind an object to a key that's not reserved leads to exception
+    @Test(expected = IllegalStateException::class)
+    fun testBindWithoutKey() {
+        val binding = TaggedWeakPeerBinding<Any>()
+        val keyNotReserved: Long = 1
+        try {
+            binding.bind(keyNotReserved, object {})
+        }
+        finally { binding.clear() }
+    }
+
+    // Rebinding an existing mapping with a different object results in exception
+    @Test(expected = IllegalStateException::class)
+    fun testRebindPeerWithDifferentObject() {
+        val binding = TaggedWeakPeerBinding<Any>()
+        try {
+            val keyReserve = binding.reserveKey()
+            val object1 = object {}
+            val object2 = object {}
+
+            binding.bind(keyReserve, object1)
+            binding.bind(keyReserve, object2)
+        }
+        finally { binding.clear() }
+    }
+
+    // Getting a binding that doesn't exist returns null
+    @Test
+    fun testGetNonExistingBinding() {
+        val binding = TaggedWeakPeerBinding<Any>()
+        try {
+            val randomNum: Long = 2
```

Comment:
why have a variable at all?  It certainly isn't a randomNum...

... and it would look funny to have `val numberTwo = 2`... ;-P